clients/src/test/java/org/apache/kafka/common/security/authenticator/ClientAuthenticationFailureTest.java

```diff
@@ -90,12 +90,12 @@ public void testConsumerWithInvalidCredentials() {
         props.put(ConsumerConfig.GROUP_ID_CONFIG, "");
         StringDeserializer deserializer = new StringDeserializer();
 
-        try (KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props, deserializer, deserializer)) {
-            assertThrows(SaslAuthenticationException.class, () -> {
-                consumer.subscribe(Collections.singleton(topic));
-                consumer.poll(Duration.ofSeconds(10));
-            });
-        }
+        KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props, deserializer, deserializer);
+        assertThrows(SaslAuthenticationException.class, () -> {
+            consumer.subscribe(Collections.singleton(topic));
+            consumer.poll(Duration.ofSeconds(10));
+            consumer.close();
+        });
```

Comment:
Why can't we use try with resource here?

Reply:
This is because the `consumer.close()` tries to send a request to the server (the final fetch request introduced in this PR) which fails with authentication exception. Hence, a try-with-resource will fail throwing out an exception. Instead I have modified the code here to expect the authentication exception.

Reply:
Seems like you'd leak the consumer if an exception is thrown by `subscribe` or `poll`? Generally, it's best to have the `assertThrows` with only the method that is meant to throw an exception.

Reply:
+1

Reply:
I have verified that this change is not needed now (since there are no sessions available to close) and hence, `consumer.close()` doesn't send a message out. I have reverted change in this test in latest commit.